kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ApplicationOperation.scala

```diff
@@ -55,21 +56,55 @@ trait ApplicationOperation {
    * Get the engine/application status by the unique application tag
    *
    * @param tag the unique application tag for engine instance.
-   * @return a map contains the application status
+   * @return [[ApplicationInfo]]
    */
-  def getApplicationInfoByTag(tag: String): Map[String, String]
+  def getApplicationInfoByTag(tag: String): ApplicationInfo
 }
 
-object ApplicationOperation {
+object ApplicationState extends Enumeration {
+  type ApplicationState = Value
+  val PENDING, RUNNING, FINISHED, KILLED, FAILED, NOT_FOUND = Value
 
-  /**
-   * identifier determined by cluster manager for the engine
-   */
-  val APP_ID_KEY = "id"
-  val APP_NAME_KEY = "name"
-  val APP_STATE_KEY = "state"
-  val APP_URL_KEY = "url"
-  val APP_ERROR_KEY = "error"
+  def fromName(name: String): ApplicationState = name.toUpperCase match {
+    // k8s has some different status string
+    case "WAITING" => PENDING
```

Comment:
where is submitted/new/accepted/failed etc

Reply:
will upadate it

Reply:
any unhandled states from kubernetes, yunicorn, or valcano etc?